### ISSUE TITLE
Crash report from users ConcurrentModiciationException

### ISSUE THREAD

**Issue by njovy** [Baseflow/flutter-geolocator#620]
## 🐛 Bug Report

I updated my app last Tuesday and I received a new crash report due to the new update. 
[GeolocationManager.java L139](https://github.com/Baseflow/flutter-geolocator/blob/geolocator_v6.1.8%2B1/geolocator/android/src/main/java/com/baseflow/geolocator/location/GeolocationManager.java#L139)
I looked into this code and locationClients can be modified and we need to add to add lock or copy this array whenever it's modified.

```
Fatal Exception: java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=8711, result=0, data=null} to activity {app.ranchat.talk/app.ranchat.talk.MainActivity}: java.util.ConcurrentModificationException
       at android.app.ActivityThread.deliverResults(ActivityThread.java:4948)
       at android.app.ActivityThread.handleSendResult(ActivityThread.java:4989)
       at android.app.servertransaction.ActivityResultItem.execute(ActivityResultItem.java:51)
       at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
       at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2058)
       at android.os.Handler.dispatchMessage(Handler.java:107)
       at android.os.Looper.loop(Looper.java:214)
       at android.app.ActivityThread.main(ActivityThread.java:7573)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:937)
```

```
Caused by java.util.ConcurrentModificationException
       at java.util.ArrayList$Itr.next(ArrayList.java:860)
       at com.baseflow.geolocator.location.GeolocationManager.onActivityResult(GeolocationManager.java:139)
       at io.flutter.embedding.engine.FlutterEnginePluginRegistry$FlutterEngineActivityPluginBinding.onActivityResult(FlutterEnginePluginRegistry.java:691)
       at io.flutter.embedding.engine.FlutterEnginePluginRegistry.onActivityResult(FlutterEnginePluginRegistry.java:378)
       at io.flutter.embedding.android.FlutterActivityAndFragmentDelegate.onActivityResult(FlutterActivityAndFragmentDelegate.java:625)
       at io.flutter.embedding.android.FlutterActivity.onActivityResult(FlutterActivity.java:583)
       at android.app.Activity.dispatchActivityResult(Activity.java:8120)
       at android.app.ActivityThread.deliverResults(ActivityThread.java:4941)
       at android.app.ActivityThread.handleSendResult(ActivityThread.java:4989)
       at android.app.servertransaction.ActivityResultItem.execute(ActivityResultItem.java:51)
       at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
       at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2058)
       at android.os.Handler.dispatchMessage(Handler.java:107)
       at android.os.Looper.loop(Looper.java:214)
       at android.app.ActivityThread.main(ActivityThread.java:7573)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:937)
```

**Version:** 6.1.8+1

**Platform:** 
- [ ] :iphone: iOS
- [x] :robot: Android

**Comment by mvanbeusekom** [comment]
@njovy thank you very much for reporting this issue. I have just released version 6.1.11 containing a fix for this exception.

**Comment by njovy** [comment]
@mvanbeusekom Wow that was super fast!! Thank you very much for this plugin and all the hard work.
